pkg/registry/core/service/ipallocator/cidrallocator.go

```diff
@@ -449,7 +449,7 @@ func (c *MetaAllocator) Destroy() {
 	select {
 	case <-c.internalStopCh:
 	default:
-		if !utilfeature.DefaultFeatureGate.Enabled(features.DisableAllocatorDualWrite) {
+		if !utilfeature.DefaultFeatureGate.Enabled(features.DisableAllocatorDualWrite) && c.bitmapAllocator != nil {
```

Comment:
if this can be nil, there's a bunch of other call sites in this file that are unprotected and look like they will panic ... can this really be nil?

Reply:
I hit this in an integration test, I think that is because the integration is not plumbing all the pieces, I imagine that if this was panicing we should have notices as it has a big impact on the apiserver

Reply:
hmm... can we fix it by initializing the object more correctly in the integration test? doing a nil check only in one place in the non-test code is pretty weird

Reply:
done

Reply:
will keep an eye on this, I do not see failures neither, maybe I touched something